File: Data Structures & Algorithms/duplicate-integer/submission-1.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:

        hash_tab = {}



        

        for i in range(len(nums)):
            if nums[i] in hash_tab:
                return True
            else:
                hash_tab[nums[i]] = i





           
        
        return False

File: Data Structures & Algorithms/two-integer-sum/submission-1.java


class Solution {
    public int[] twoSum(int[] nums, int target) {

        HashMap<Integer, Integer> hashes = new HashMap<>();


        for(int i = 0; i<nums.length; i++){
            hashes.put(nums[i], i);
        }


        for(int i=0; i<nums.length; i++){
            int different = target - nums[i];
            if(hashes.containsKey(different) && hashes.get(different) != i){
                return new int[]{i, hashes.get(different)};
            }
        }


        return new int[0];
    }
}
